File: LINKED_LIST/sameTree.c
#include<stdio.h>
#include<stdlib.h>
#include "create_tree_node.h"
#include<stdbool.h>

bool sametree(struct node *root1,struct node *root2)
{
  if(root1==NULL && root2== NULL)
  return true;
  if(root1==NULL || root2==NULL)
  return false;

  return (root1->data==root2->data) && sametree(root1->left,root2->left) && sametree(root1->right,root2->right);

}

void main()
{
  struct node *root1=NULL;
  struct node *root2=NULL;

  root1=create_node(10);
  root1->left=create_node(16);
  root1->right=create_node(15);
  root1->right->left=create_node(18);

  root2=create_node(10);
  root2->left=create_node(16);
  root2->right=create_node(15);
  root2->right->left=create_node(33);

  bool d=sametree(root1,root2);
  if(d==true)
  printf("same tree\n");
  else
  printf("tree is not same \n");


}

File: LINKED_LIST/JAVA/Array/SameTree.java
import java.util.*;
import java.io.*;

class Node
{
    int data;
    Node left;
    Node right;
    Node(int item)
    {
        data=item;
        left=null;
        right=null;
    }
}
    class SameTree
    {   Node root;
        Node root2;
        SameTree()
        {
            root=null;
            root2=null;
        }

        boolean sameTreeUtil(Node node1, Node node2)
        {
            if(node1==null && node2==null)
            {
                return true;
            }

            if(node1!=null && node2!=null)
            {
                return((node1.data==node2.data) && sameTreeUtil(node1.left,node2.left) && sameTreeUtil(node1.right,node2.right));


            }
            else
                return false;
        }

        boolean HelperUtil()
        {
            return sameTreeUtil(root,root2);
        }

    public static void main(String args[])
    {
        SameTree tree1=new SameTree();


        tree1.root= new Node(2);
        tree1.root.right=new Node(3);
        tree1.root.left=new Node(4);

        tree1.root2= new Node(2);
        tree1.root2.right=new Node(3);
        tree1.root2.left=new Node(4);

        boolean b=tree1.HelperUtil();
        if(b==true)
            System.out.println("Sametree");
        else
            System.out.println("not a Sametree");

    }

}
